src/Formatter/CurlCommandFormatter.php

```diff
@@ -42,9 +42,12 @@ public function formatRequest(RequestInterface $request)
             } elseif ($body->isSeekable()) {
                 $data = $body->__toString();
                 $body->rewind();
-                if (preg_match('/[\x00-\x1F\x7F]/', $data)) {
+                // all non-printable ASCII characters and <DEL> except for \t, \r, \n
+                if (preg_match('/([\x00-\x09\x0C\x0E-\x1F\x7F])/', $data)) {
                     $data = '[binary stream omitted]';
                 }
+                // replace \t, \r and \n with their string literals
+                $data = str_replace(["\t", "\r", "\n"], ['\t', '\r', '\n'], $data);
```

Comment:
if i copy-paste this onto a console, does the `\t` / `\r` / `\n` work correctly? i am a bit afraid of cluttering the content with such characters - but when we output as html, the newlines would get lost, so this might be the right solution.

Reply:
Good catch, it will not work due to the `escapeshellarg()` later on. We could simple not care about these three characters at all and `escapeshelarg()` will do the rest. The output will then look somehow like this:

```
curl --......  --data 'foo
bar' http://foo/bar
```

This doesn't look nice, but it can be copy and pasted.

Reply:
i prefer less nice but working :-D